### DISCUSSION TITLE
context.NewCommand and ReadData

### DISCUSSION THREAD

**Discussion by angelru** [Megabit/Blazorise#2186] [Q&A] [answered]
I am using ReadData and @ context.NewCommand.Clicked

when I save the new data ReadData runs again, is it proper behavior? because of course I have my api call without the add call, and it is replaced with my api returns, then the addition is removed

**Comment by stsrki** [comment, marked as answer]
Yes, calling ReadData after save it the proper behavior.

**Reply by angelru** [reply]
I don't understand, but what is the point of calling it automatically after saving? so I have to replace `@context.NewCommand.Clicked` with my save method?

**Reply by stsrki** [reply]
Not sure what you mean by replacing NewCommand with the save method. NewCommand only changes the state of the grid so that you can go into the edit mode. Only when you click on Save does the actual save happens and ReadData is called.

And it is called because your in-memory data and the real data somewhere in DB can differ. For example, you can only edit few fields in the model, while others are populated from your backend. It is a lot easier to always call the ReadData so that DataGrid can be repopulated with real live data.

In any case, if you don't want to call your backend you can just set some boolean flag on the Save command that will let you skip reload in ReadData if you wish it that way.
